src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+    this->unk_26E = 1;
+    this->unk_211 = false;
+    this->isAwake = false;
+    this->actionFunc = EnGo2_CurledUp;
+}
+
+void EnGo2_WakeUp(EnGo2* this, GlobalContext* globalCtx) {
+    if (this->skelAnime.animPlaybackSpeed == 0.0f) {
+        if ((this->actor.params & 0x1F) != GORON_DMT_BIGGORON) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOLON_WAKE_UP);
+        } else {
+            func_800F4524(&D_801333D4, NA_SE_EN_GOLON_WAKE_UP, 60);
+        }
+    }
+    if ((this->actor.params & 0x1F) == GORON_DMT_BIGGORON) {
+        func_800800F8(globalCtx, 0x1068, -0x63, &this->actor, 0);
+        func_80034EC0(&this->skelAnime, sAnimations, 10);
+        this->skelAnime.animPlaybackSpeed = 0.5f;
+    } else {
+        func_80034EC0(&this->skelAnime, sAnimations, 1);
+        this->skelAnime.animPlaybackSpeed = 1.0f;
+    }
+    this->actionFunc = func_80A46B40;
+}
+
+void EnGo2_GetItemAnimation(EnGo2* this, GlobalContext* globalCtx) {
+    func_80034EC0(&this->skelAnime, sAnimations, 1);
+    this->unk_211 = true;
+    this->actionFunc = func_80A46B40;
+    this->skelAnime.animPlaybackSpeed = 0.0f;
+    this->actor.speedXZ = 0.0f;
+    this->skelAnime.animCurrentFrame = this->skelAnime.animFrameCount;
+}
+
+void EnGo2_BeginRolling(EnGo2* this, GlobalContext* globalCtx) {
+    if ((this->actor.params & 0x1F) == GORON_CITY_ROLLING_BIG || (this->actor.params & 0x1F) == GORON_CITY_LINK) {
+        this->collider.body.bumperFlags = 1;
+        if (gSaveContext.infTable[17] & 0x4000) {
+            this->actor.speedXZ = 6.0f;
+        } else {
+            this->actor.speedXZ = 3.6000001f;
+        }
+    } else {
+        this->actor.speedXZ = 6.0f;
+    }
+    this->actor.flags |= 0x1000000;
+    this->animTimer = 10;
+    this->actor.shape.unk_08 = 1800.0f;
+    this->actor.speedXZ += this->actor.speedXZ; // Speeding up
+    this->actionFunc = EnGo2_ContinueRolling;
+}
+
+void EnGo2_StopRolling(EnGo2* this, GlobalContext* globalCtx) {
+    EnBom* bomb;
+
+    if (((this->actor.params & 0x1F) != GORON_CITY_ROLLING_BIG) && ((this->actor.params & 0x1F) != GORON_CITY_LINK)) {
+        if ((this->actor.params & 0x1F) == GORON_DMT_ROLLING_SMALL) {
+            bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.posRot.pos.x,
+                                       this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 0);
+            if (bomb != NULL) {
+                bomb->timer = 0;
+            }
+        }
+    } else {
+        this->collider.body.bumperFlags = 0;
+    }
+
+    this->actor.shape.rot = this->actor.posRot.rot;
+    this->unk_59C = 0;
+    this->unk_590 = 0;
+    this->actionFunc = EnGo2_GroundRolling;
+    this->actor.shape.unk_08 = 0.0f;
+    this->actor.speedXZ = 0.0f;
+}
+
+s32 EnGo2_IsFreeingGoronInFire(EnGo2* this, GlobalContext* globalCtx) {
+    if ((this->actor.params & 0x1F) != GORON_FIRE_GENERIC) {
+        return false;
+    } else {
```

Comment:
Same as an earlier comment, you can remove this `else` thanks to the early return